src/operation/decision_model_optimizer_results.jl

```diff
@@ -0,0 +1,84 @@
+"""
+Stores results data for one DecisionModel
+"""
+mutable struct DecisionModelOptimizerResults <: AbstractModelOptimizerResults
+    duals::Dict{ConstraintKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}}
+    parameters::Dict{ParameterKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}}
+    variables::Dict{VariableKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}}
+    aux_variables::Dict{AuxVarKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}}
+    expressions::Dict{ExpressionKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}}
+    optimizer_stats::OrderedDict{Dates.DateTime, OptimizerStats}
+end
+
+function DecisionModelOptimizerResults()
+    return DecisionModelOptimizerResults(
+        Dict{
+            ConstraintKey,
+            Dict{ConstraintKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}},
+        }(),
+        Dict{
+            ParameterKey,
+            Dict{ParameterKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}},
+        }(),
+        Dict{
+            VariableKey,
+            Dict{VariableKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}},
+        }(),
+        Dict{
+            AuxVarKey,
+            Dict{ConstraintKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}},
+        }(),
+        Dict{
+            AuxVarKey,
+            Dict{ExpressionKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}},
+        }(),
+        OrderedDict{Dates.DateTime, OptimizerStats}(),
+    )
+end
+
+function write_result!(
```

Comment:
@daniel-thom should we support an interface that doesn't require passing the field and we choose it based on the `OptimizationContainerKey`?

Reply:
We could add the interface that chooses the correct field based on the key type in the abstract

Reply:
Refer to the functions in abstract_model_optimizer_results. I handled it there.